### PR TITLE
fix(windows): autoInstallOnAppQuit

### LINKED ISSUE #1570: Auto-update broken on Windows
This issue is directly related to issue #1514 .  Not sure why this was closed before verifying that it worked, because, when I initiated the auto update to ipfs-desktop 12.0, the exact same issue occurred once again. Program was uninstalled, not re-installed, and the installation files were located in the %USERPROFILE%\Appdata\local\ipfs-desktop-updater. I did notice that there were 2 executable files in the directory though (I don't recall if there were last time): one named "updater.exe" and one that was the actual install file for the program (IPFS-Desktop-Setup-0.12.0.exe). Interestingly, while both were the same size, (83,195 kb), I noticed that the hash values were different, so they are not the same file? I don't know. What I do know is that the process is still broken and the potential cause of the issue which I mentioned in the 2nd of the 2 potential causes I listed remains unchanged (update-info.json "isAdminRightsRequired" value is still euqal to false, and as I stated the first time, you need admin permissions to install an application to the program files directory, which is where ipfs desktop is installed when you install it for all users). Maybe I am an outlier in this mindset, but I am of the belief that issues should be confirmed resolved before closing them.Auto update not working on Windows when installed globally for all users

 - OS: Windows 10
 - Version of IPFS Desktop [e.g. 0.12.0]

**Describe the bug**
I have noticed that, the last two times that IPFS-desktop has updated using the internal update functionality (when the app says a new version is available and to click button to close IPFS-desktop and install the updated version), the old version of ipfs-desktop successfully uninstalls the old version, but does not install the new version **correctly.**  I should note that I install IPFS-desktop so that it available for all users, not just the currently running user, which I think is the main reason the auto-update might be failing.  There are a few reasons that I can think of that might be the cause:

1. In my case, when I installed IPFS, I installed it so that it would be available for all users.  As you may know, this is not the default configuration for the installer (the default being to install IPFS-desktop only for the current user), so what **MAY** be happening is that IPFS-desktop is uninstalling the old version located in Program files (the folder to install app for all users), but then installing the new version in app data (only install for current user).  If this turns out to be the case, while technically still successfully installing the application (which is why I emphasized "correctly"), my IPFS-desktop Desktop icon can no longer find its source executable, and I just end up downloading the installer and reinstalling the application manually anyway, defeating the auto update functionality entirely.  This is unlikely the cause.
2. The process initiating that auto-update functionality may not have the elevated permissions that it needs to write to the Program files directory.  If the installer takes into account the install settings from the previous installation (in other words, installs the new version for all users if the previous install was configured to be installed for all users), then the reason that it might be failing is because the process may lack the elevated permissions needed to write to the Program Files directory.  I am confident that this is the cause of the failed installation, as, when I navigated to %USERNAME%\appdata\local\ipfs-desktop-updater\pending and opened the update-info.json file, I saw the following:

image

**To Reproduce**
1. Install ipfs-desktop version 0.11.2 or 0.11.1 and configure the installation so that the application is available to all users
2. Launch the application
3. Wait for the auto-update functionality dialog box to appear and initiate the process

**Expected behavior**
IPFS auto-update functionality would automatically re-install the application successfully with the same configurations as the previous install

[error.log](https://github.com/ipfs-shipyard/ipfs-desktop/files/4646202/error.log)

### LINKED ISSUE #1630: IPFS Desktop stuck at "IPFS is Starting" on Windows 10
Please describe what you were doing when this error happened.

Just trying to install and use it.  Stuck on "IPFS is Starting".  Tried rebooting.  Tried reinstalling.  Tried updating.  When I tried to update, it crashed with the following:

**Specifications**

- **OS**: win32
- **IPFS Desktop Version**: 0.12.2
- **Electron Version**: 8.2.5
- **Chrome Version**: 80.0.3987.165

**Error**

```
TypeError: ctx.checkForUpdates is not a function
    at click (C:\Program Files\IPFS Desktop\resources\app.asar\src\tray.js:198:30)
    at MenuItem.click (electron/js2c/browser_init.js:1501:9)
    at Function.executeCommand (electron/js2c/browser_init.js:1762:13)
```

### PR DESCRIPTION
**Pull request by lidel**
https://github.com/lidel/electron-updater-altschmerz/issues/1 demonstrated that `autoInstallOnAppQuit` works fine on Windows, but we had a bunch of code responsible for doing it manually, probably due to macOS not being supported natively by `autoInstallOnAppQuit`.

Docs for `electron-updater` 4.3.5 state:
>  Screenshot_2020-10-09 Auto Update - electron-builder

Due to this, the fix from https://github.com/ipfs-shipyard/ipfs-desktop/pull/1679 was not applied correctly: macOS-specific handling executed first and broke updates on Windows.

This PR ensures that macOS keeps using the old upgrade paths, but Windows and AppImage leverage upstream code for applying upgrades.

Closes #1570 and most likely closes #1630  too.

### CODE REVIEW

**Comment by lidel** [comment]
Sadly this last mile is specific to how our codebase and won't fix old versions.

To properly test this we need to tag v0.13.1 and v0.13.2 and do quick smoke test by installing v0.13.1 and testing if it upgrades fine to v0.13.2. 

Autoupdate on Windows does not require signing, so I'll try to do the above test in a fork first.

**Comment by lidel** [comment]
Ok, tested this on Windows with my fork and both per user and per all users upgraded correctly on Windows.
There should be no change to behaviour on macOS.

File: src/daemon/index.js [hunk 1]
```diff
@@ -109,12 +109,8 @@ module.exports = async function (ctx) {
 
   ipcMain.on('ipfsConfigChanged', restartIpfs)
 
-  app.on('before-quit', async e => {
-    if (ipfsd) {
-      e.preventDefault()
-      await stopIpfs()
-      app.quit()
```
Comment (lidel) [inline src/daemon/index.js:116]: not sure what was the purpose of this, but it seemed to interfere with `autoInstallOnAppQuit
` on Windows

File: src/auto-updater/index.js [hunk 2]
```diff
@@ -4,13 +4,17 @@ const i18n = require('i18next')
 const logger = require('../common/logger')
 const { notify } = require('../common/notify')
 const { showDialog } = require('../dialogs')
-const quitAndInstall = require('./quit-and-install')
+const macQuitAndInstall = require('./macos-quit-and-install')
+const macOS = process.platform === 'darwin'
```
Comment (hacdias) [inline src/auto-updater/index.js:8]: Just a nitpick: we already have this constants defined on [`../common/consts`](https://github.com/ipfs-shipyard/ipfs-desktop/blob/master/src/common/consts.js).

**Review by hacdias — APPROVED** [review]
LGTM!

**Pull request merged**